src/main/java/gov/nasa/pds/harvest/dao/MetadataWriter.java

```diff
@@ -111,9 +113,34 @@ private void writeBatch() throws Exception
             }
         }
         
-        totalRecords += loader.loadBatch(data);
+        // Load batch
+        Set<String> failedIds = new TreeSet<>();
+        totalRecords += loader.loadBatch(data, failedIds);
         log.info("Wrote " + totalRecords + " product(s)");
         
+        // Update failed counter
+        counter.failedFileCount += failedIds.size();
+
+        // Update product counters
+        for(RegistryDocBatch.NJsonItem item: docBatch.getItems())
+        {
+            if(nonRegisteredIds == null)
```

Comment:
this statement and the following could be on one line, with a boolean operator in between the 2 conditions